modules/command-engine/core/src/main/scala/surge/core/commondsl/SurgeCommandBusinessLogic.scala

```diff
@@ -2,75 +2,35 @@
 
 package surge.core.commondsl
 
-import com.typesafe.config.Config
-import io.opentracing.Tracer
-import io.opentracing.noop.NoopTracerFactory
 import surge.core.command.{ AggregateCommandModelCoreTrait, SurgeCommandKafkaConfig }
-import surge.core.{ SurgeAggregateReadFormatting, SurgeWriteFormatting }
+import surge.core.{ SurgeAggregateReadFormatting, SurgeAggregateWriteFormatting, SurgeEventWriteFormatting, SurgeWriteFormatting }
+import surge.internal.kafka.SurgeKafkaConfig
 import surge.kafka.KafkaTopic
-import surge.metrics.Metrics
 
-trait SurgeGenericBusinessLogicTrait[AggId, Agg, Command, Rej, Event] {
+trait SurgeGenericCommandBusinessLogicTrait[AggId, Agg, Command, Rej, Event] extends SurgeGenericBusinessLogicTrait[AggId, Agg, Command, Rej, Event] {
 
-  protected[surge] val config: Config
-
-  def aggregateName: String
-
-  def stateTopic: KafkaTopic
   def eventsTopic: KafkaTopic
-
-  def publishStateOnly: Boolean
-
-  def readFormatting: SurgeAggregateReadFormatting[Agg]
-
   def writeFormatting: SurgeWriteFormatting[Agg, Event]
+  def readFormatting: SurgeAggregateReadFormatting[Agg]
```

Comment:
These names and aggregations are a little wonky, but I kept them to maintain compatibility with existing code.